packages/fxa-auth-server/lib/payments/subscription-reminders.ts

```diff
@@ -0,0 +1,216 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Logger } from 'mozlog';
+import Stripe from 'stripe';
+import { Container } from 'typedi';
+
+import { ConfigType } from '../../config';
+import { reportSentryError } from '../sentry';
+import { SentEmailParams, Plan } from 'fxa-shared/subscriptions/types';
+import { StripeHelper, TimeSpanInS } from './stripe';
+import { SentEmail } from 'fxa-shared/db/models/auth';
+
+import {
+  DEFAULT_PLAN_LENGTH,
+  DEFAULT_REMINDER_LENGTH,
+} from '../../scripts/subscription-reminders';
+
+export const EMAIL_TYPE: string = 'subscriptionRenewalReminder';
+const DAYS_IN_A_WEEK: number = 7;
+const DAYS_IN_A_MONTH: number = 30;
+const DAYS_IN_A_YEAR: number = 365;
+export const MS_IN_A_DAY: number = 24 * 60 * 60 * 1000;
+
+interface EligiblePlansByInterval {
+  day(plan: Plan, planLength: number): Boolean;
+  week(plan: Plan, planLength: number): Boolean;
+  month(plan: Plan, planLength: number): Boolean;
+  year(plan: Plan, planLength: number): Boolean;
+}
+
+const eligiblePlansByInterval: EligiblePlansByInterval = {
+  day: (plan, planLength) => plan.interval_count >= planLength,
+  week: (plan, planLength) =>
+    plan.interval_count >= planLength / DAYS_IN_A_WEEK,
+  month: (plan, planLength) =>
+    plan.interval_count >= planLength / DAYS_IN_A_MONTH,
+  year: (plan, planLength) =>
+    plan.interval_count >= planLength / DAYS_IN_A_YEAR,
+};
+
+export class SubscriptionReminders {
+  private db: any;
+  private mailer: any;
+  private stripeHelper: StripeHelper;
+
+  constructor(
+    private log: Logger,
+    config: ConfigType,
+    private planLength: number = DEFAULT_PLAN_LENGTH,
+    private reminderLength: number = DEFAULT_REMINDER_LENGTH,
+    db: any,
+    mailer: any
+  ) {
+    this.db = db;
+    this.mailer = mailer;
+    this.stripeHelper = Container.get(StripeHelper);
+  }
+
+  /**
+   * For all possible plan.intervals, determine if the plan is sufficiently
+   * long based on planLength.
+   */
+  private isEligiblePlan(plan: Plan): boolean {
+    for (let [interval, checkFn] of Object.entries(eligiblePlansByInterval)) {
+      if (plan.interval === interval && checkFn(plan, this.planLength)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private async getEligiblePlans(): Promise<Plan[]> {
+    const allPlans = await this.stripeHelper.allPlans();
+    return allPlans.filter((plan) => this.isEligiblePlan(plan));
+  }
+
+  /**
+   * Returns a window of time in seconds [startTimeS, endTimeS)
```

Comment:
`[` not matching `)`, but did you mean `{}` to match the return type?

Reply:
I meant that the window of time was inclusive of the lower bound and exclusive of the upper bound. It's a convention I picked up in some math classes in college. Looking at this method, that information is actually relevant to the `findActiveSubscriptionsByPlanId`, not this one, so I'll change these to `{` and `}`.